pkg/descheduler/descheduler.go

```diff
@@ -153,8 +153,65 @@ func metricsProviderListToMap(providersList []api.MetricsProvider) map[api.Metri
 	return providersMap
 }
 
+// preserveNeeded returns the obj preserving fields needed for memory efficiency.
+// Only keep scheduler related fields
+func preserveNeeded(obj interface{}) (interface{}, error) {
+	if pod, ok := obj.(*v1.Pod); ok {
+		preserveContainer := func(c *v1.Container) {
+			c.Command = nil
+			c.Args = nil
+			c.WorkingDir = ""
+			c.Ports = nil
+			c.EnvFrom = nil
+			c.Env = nil
+			c.ResizePolicy = nil
+			c.VolumeMounts = nil
+			c.VolumeDevices = nil
+			c.LivenessProbe = nil
+			c.ReadinessProbe = nil
+			c.StartupProbe = nil
+			c.Lifecycle = nil
+			c.TerminationMessagePath = ""
+			c.TerminationMessagePolicy = ""
+			c.ImagePullPolicy = ""
+			c.SecurityContext = nil
+		}
+
+		// metadata related
+		pod.ManagedFields = nil
+		pod.Finalizers = nil
+
+		// spec related
+		for i := 0; i < len(pod.Spec.InitContainers); i++ {
+			preserveContainer(&pod.Spec.InitContainers[i])
+		}
+		for i := 0; i < len(pod.Spec.Containers); i++ {
+			preserveContainer(&pod.Spec.Containers[i])
+		}
+		pod.Spec.EphemeralContainers = nil
+		pod.Spec.SecurityContext = nil
+		pod.Spec.ImagePullSecrets = nil
+		pod.Spec.Hostname = ""
+		pod.Spec.Subdomain = ""
+		pod.Spec.HostAliases = nil
+		pod.Spec.PriorityClassName = ""
+		pod.Spec.DNSConfig = nil
+		pod.Spec.ReadinessGates = nil
+		pod.Spec.RuntimeClassName = nil
+		pod.Spec.PreemptionPolicy = nil
+	}
+	if node, ok := obj.(*v1.Node); ok {
+		node.ManagedFields = nil
```

Comment:
this is already being done [here](https://github.com/kubernetes-sigs/descheduler/blob/94d1825c6866623469bf3eaec1b34ada7504d6ae/pkg/descheduler/descheduler.go#L715-L720)